src/libraries/System.Text.RegularExpressions/src/System/Text/PositionStringBuilder.cs

```diff
@@ -0,0 +1,103 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System.Text
+{
+    /// <summary>Provides a value type string builder composed of individual segments represented as <see cref="ReadOnlyMemory{T}"/> instances.</summary>
+    [DebuggerDisplay("Count = {_count}")]
+    internal struct OffsetCountStringBuilder
```

Comment:
Nit: the file name doesn't match the type name

Could this be replaced by `ValueListBuilder` to avoid some of the duplication, or do you expect it may be used in more places in the future?

Reply:
ValueListBuilder is a ref struct. It can't be used here until we have the ability to pass ref structs in generics.

Reply:
Done